enterprise/cypher/morsel-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/vectorized/PipelineBuilder.scala

```diff
@@ -102,6 +103,26 @@ class PipelineBuilder(slotConfigurations: SlotConfigurations, converters: Expres
           source = source.addOperator(preSorting)
           new MergeSortOperator(ordering, slots)
 
+        case plans.Aggregation(_, groupingExpressions, aggregationExpression) if groupingExpressions.isEmpty =>
+          val (in, out) = aggregationExpression.foldLeft[(Seq[MapperOffsets], Seq[ReducerOffsets])]((Seq.empty, Seq.empty)) {
+            case ((maps, reds), (key, expression)) =>
+              val aggregator = converters.toCommandExpression(expression).asInstanceOf[AggregationExpressionOperator]
+              val currentSlot = slots.get(key).get
+              //we need to make room for storing aggregation value in
+              //source slot
+              source.slots.newReference(key, currentSlot.nullable, currentSlot.typ)
```

Comment:
Hmmm... If I understand this correctly, what we'll do is to use the input morsel first to calculate the aggregate values, and then we write those values back into the same morsel. Could we not instead break the pipeline here? From what I can see, the result will be that the input-morsel will be wider than necessary (since the aggregation values cannot be written until after all the input has been consumed), and for the output, we only use the last ref-slots of the first row.

It should work but seems inefficient.